bindings/event_driven.i

```diff
@@ -0,0 +1,223 @@
+                                            (unsigned int* s2, int m2),
+                                            (unsigned int* s3, int m3),
+                                            (unsigned int* s4, int m4),
+                                            (unsigned int* s5, int m5)};
+
+namespace ev {
+
+
+class vEvent
+{
+public:
+    static const std::string tag;
+    vEvent();
+    virtual ~vEvent();
+
+    virtual void encode(yarp::os::Bottle &b) const;
+    virtual bool decode(const yarp::os::Bottle &packet, int &pos);
+    virtual yarp::os::Property getContent() const;
+    virtual std::string getType() const;
+    virtual int getChannel() const;
+    virtual void setChannel();
+};
+
+
+class AddressEvent : public vEvent
+{
+public:
+    AddressEvent();
+    AddressEvent(const vEvent &v);
+    AddressEvent(const AddressEvent &v);
+
+    virtual void encode(yarp::os::Bottle &b) const;
+    virtual bool decode(const yarp::os::Bottle &packet, int &pos);
+    virtual yarp::os::Property getContent() const;
+    virtual std::string getType() const;
+    virtual int getChannel() const;
+    virtual void setChannel(const int channel);
+};
+class FlowEvent : public AddressEvent
+{
+public:
+    static const std::string tag;
+    float vx;
+    float vy;
+
+    FlowEvent();
+    FlowEvent(const vEvent &v);
+    FlowEvent(const FlowEvent &v);
+
+    virtual void encode(yarp::os::Bottle &b) const;
+    virtual bool decode(const yarp::os::Bottle &packet, int &pos);
+    virtual yarp::os::Property getContent() const;
+    virtual std::string getType() const;
+
+    int getDeath() const;
+};
+
+class LabelledAE : public AddressEvent
+{
+public:
+    static const std::string tag;
+    int ID;
+
+    LabelledAE();
+    LabelledAE(const vEvent &v);
+    LabelledAE(const LabelledAE &v);
+
+    virtual void encode(yarp::os::Bottle &b) const;
+    virtual bool decode(const yarp::os::Bottle &packet, int &pos);
+    virtual yarp::os::Property getContent() const;
+    virtual std::string getType() const;
+};
+
+class GaussianAE : public LabelledAE
+{
+public:
+    static const std::string tag;
+    float sigx;
+    float sigy;
+    float sigxy;
+
+    GaussianAE();
+    GaussianAE(const vEvent &v);
+    GaussianAE(const GaussianAE &v);
+
+    virtual void encode(yarp::os::Bottle &b) const;
+    virtual bool decode(const yarp::os::Bottle &packet, int &pos);
+    virtual yarp::os::Property getContent() const;
+    virtual std::string getType() const;
+};
+
+class vBottle : public yarp::os::Bottle {
+
+public:
+
+    //constructors should not change from Bottle
+    vBottle();
+
+    //you can only modify contents by adding events and append other vBottles
+    void append(vBottle &eb);
+
+    template<class T> void append(vBottle &eb);
+
+    //inherited
+    void clear();
+
+    bool read(yarp::os::ConnectionReader& reader);
+
+
+
+
+
+};
+
+
+%extend vBottle {
+    int getSize() {
+        auto q = $self->get<ev::vEvent>();
```

Comment:
this seems like an inefficient way to calculate the size. perhaps we can improve it, especially if it is used often